code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -137,13 +166,81 @@
 	regular_hud_updates()
 	standard_damage_overlay_updates()
 
+
+	if((stat!=DEAD) && (stat!=UNCONSCIOUS) && !busy && !ismoulting && !client && !mind && ckey==null) //Checks for AI
+		grue_ai()
+
+
+
+
+
+//AI stuff:
+/mob/living/simple_animal/hostile/grue/proc/grue_ai()
+
+	//Moulting
+	if(lifestage!=GRUE_ADULT && (nutrienergy>=moultcost) && dark_dim_light==GRUE_DARK)
+		start_moult()
+
+	//Eating
+	if(lifestage>=GRUE_JUVENILE && stance==HOSTILE_STANCE_IDLE && dark_dim_light<GRUE_LIGHT)
+		var/list/feed_targets = list()
+		for(var/mob/living/carbon/C in range(1,get_turf(src)))
+			feed_targets += C
+		if(feed_targets.len)
+			handle_feed(pick(feed_targets))
+
+	//Egglaying
+	if(lifestage==GRUE_ADULT && eatencharge>0 && dark_dim_light==GRUE_DARK)
+		reproduce()
+
+	//Movement
+	//
+	//Wandering behavior
+	if((!client||deny_client_move) && !busy && !ismoulting && !anchored && (ckey == null) && !(flags & INVULNERABLE))
+		if(canmove)
+			turns_since_move++
+			if(turns_since_move >= turns_per_move)
+				if(!(stop_automated_movement_when_pulled && pulledby))
+					INVOKE_EVENT(src, /event/before_move)
+					var/list/potential_dests=list() //any potential wander destinations
+					var/list/respite_dests=list() //any potential wander destinations leading out of the light into dark/dim
+					for(var/thiscardinal in 1 to cardinal.len)
+						var/potential_dest = get_step(src, cardinal[thiscardinal])
+						if(dark_dim_light==GRUE_LIGHT)//always prioritize wandering to a dark/dim tile if in light
+							if(get_ddl(potential_dest)<GRUE_LIGHT)
+								respite_dests+=potential_dest
+							else
+								potential_dests+=potential_dest
+						else //never wander from a dark/dim tile into light
+							if(get_ddl(potential_dest)<GRUE_LIGHT)
+								potential_dests+=potential_dest
+					var/ourdest
+					if(respite_dests.len)
+						ourdest=(pick(respite_dests))
+					else if(potential_dests.len)
+						ourdest=(pick(potential_dests))
+					if(ourdest)
+						wander_move(ourdest)
+						turns_since_move = 0
+						INVOKE_EVENT(src, /event/after_move)
```

Comment:
`wander_move` will call `Move` which should invoke this event. Is this necessary for some reason?

Reply:
I copied it along from elsewhere, I'll remove it.